### PR TITLE
feat: getPermissionStatus

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### Summary
This PR adds a new `getPermissionStatus` utility function that allows developers to check the real-time status of spend permissions, including remaining allowance, next period timing, and activation status.

#### New Features
- **`getPermissionStatus` method**: Query blockchain to get real-time spend permission status
  - Returns remaining spend allowance for current period
  - Provides next period start timestamp
  - Indicates whether permission is still active (not revoked and valid)
- **`toSpendPermissionArgs` utility**: Convert SpendPermission objects to contract-compatible arguments
  - Handles address checksumming via viem
  - Converts string values to appropriate BigInt types
  - Maintains proper field ordering for contract calls

#### Implementation Details
- Leverages existing chain client infrastructure via `getClient()`
- Makes parallel contract calls to SpendPermissionManager for optimal performance:
  - `getCurrentPeriod()` - get current spending period info
  - `isRevoked()` - check if permission was revoked
  - `isValid()` - check if permission is still valid
- Comprehensive error handling for missing chainId and unavailable clients
- Full TypeScript support with proper type definitions

#### Testing
- Added comprehensive test suite covering:
  - Successful status retrieval scenarios
  - Error handling (missing chainId, unavailable client, contract errors)
  - Edge cases (zero allowance, large values, exceeded spending)
  - Contract call verification and parallel execution
- Added utility function tests for address checksumming and type conversions

#### Developer Experience
- Exported in public API for easy importing
- Added to test app window object for convenient testing
- Detailed JSDoc documentation with usage examples

### Usage Example
```typescript
import { getPermissionStatus } from '@base-org/account/spend-permission';

const status = await getPermissionStatus(permission);
console.log(`Remaining: ${status.remainingSpend} wei`);
console.log(`Next period: ${new Date(parseInt(status.nextPeriodStart) * 1000)}`);
console.log(`Active: ${status.isActive}`);
```

### Dependencies
Builds on existing spend permission infrastructure from the base branch without introducing new external dependencies.

### _How did you test your changes?_

Screenshot 2025-08-01 at 1 45 17 PM

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by spencerstock — COMMENTED** [review]
Nice! 1 comment

File: packages/account-sdk/src/interface/public-utilities/spend-permission/utils.ts [hunk 1]
```diff
@@ -116,3 +117,56 @@ function getRandomHexString(byteLength: number): `0x${string}` {
...
+ * @returns A Date object.
+ */
+export function fromTimestampInSeconds(timestamp: number): Date {
+  return new Date(timestamp * 1000);
+}
+
+/**
+ * Converts a SpendPermission object to the arguments expected by the SpendPermissionManager contract.
+ *
+ * The order of the parameters does not matter because the contract will sort them.
```
[the remark this answers is not in the document]
Comment (fan-zhang-sv) [reply packages/account-sdk/src/interface/public-utilities/spend-permission/utils.ts]: yuppp updateddd

File: packages/account-sdk/src/interface/public-utilities/spend-permission/utils.ts [hunk 2]
```diff
@@ -113,6 +114,64 @@ function getRandomHexString(byteLength: number): `0x${string}` {
...
+    account: getAddress(account),
+    spender: getAddress(spender),
+    token: getAddress(token),
+    allowance: BigInt(allowanceStr),
+    period,
+    start,
+    end,
+    salt: BigInt(salt),
+    extraData: extraData as Hex,
+  };
```
Comment (spencerstock) [inline packages/account-sdk/src/interface/public-utilities/spend-permission/utils.ts:176]: TY! Looks proper.

**Review by spencerstock — APPROVED** [review]
LGTM
